sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -1064,6 +1064,34 @@ class FileSourceCharVarcharTestSuite extends CharVarcharTestSuite with SharedSpa
       }
     }
   }
+
+  test("SPARK-48498: always do char padding in predicates") {
+    import testImplicits._
+    withSQLConf(SQLConf.READ_SIDE_CHAR_PADDING.key -> "false") {
```

Comment:
cloud you add a test for conf `LEGACY_NO_CHAR_PADDING_IN_PREDICATE`?
It maybe would fail when `LEGACY_NO_CHAR_PADDING_IN_PREDICATE` = true
https://github.com/apache/spark/pull/46832/files?diff=split&w=1#r1642368622

Reply:
It will fail, as the legacy behavior is wrong. This is why we make this fix...

Reply:
got it, thanks~